File: mobile_app/app/src/main/java/com/example/licenta_mobile/security/Token.java
package com.example.licenta_mobile.security;

import com.auth0.android.jwt.Claim;
import com.auth0.android.jwt.JWT;
import com.example.licenta_mobile.model.UserData;

public class Token {
    private static String jwtToken;

    public static void setJwtToken(String token) {
        JWT parsedJWT = new JWT(token);
        Claim userId = parsedJWT.getClaim("jti");
        UserData.setUserId(userId.asInt());
        jwtToken = token;
    }

    public static String getJwtToken() {
        return jwtToken;
    }

}

File: mobile_app/app/src/main/java/com/example/licenta_mobile/security/Token.kt
package com.example.licenta_mobile.security

import com.auth0.android.jwt.JWT
import com.example.licenta_mobile.model.UserData

object Token {
    var jwtToken: String? = null
        set(token) {
            val parsedJWT = JWT(token!!)
            val userId = parsedJWT.getClaim("jti")
            UserData.userId = userId.asInt()!!
            field = token
        }
}